public/sketch.js

```diff
@@ -1,116 +1,82 @@
-}
+async function setup() {
+  noCanvas();
+  await countVotes();
+  setInterval(countVotes, 500);
 
-let firstRender = true;
+  let pollQ = createElement('p', poll.question);
+  pollQ.addClass("question");
 
-async function countVotes() {
-  // TODO this page should be for a specific poll
-  poll_id = getPollID();
-  const response = await fetch(`/poll/${poll_id}`);
-  poll = await response.json();
-  if (poll.message) {
-    throw new Error(poll.message);
+  radio = createRadio();
+  for (let i = 0; i < poll.options.length; i++) {
+    radio.option(i, poll.options[i]); // First arg is index, second arg is what is visible to user 
   }
-  displayResults(poll);
+  // radio.style('width', '50px'); // Change this for width of radio 
+
+  // Create the vote button
+  voteButton = createButton('vote');
+  // And add a class and a mouse pressed event listener
+  voteButton.addClass("VoteBTN");
+  voteButton.mousePressed(submitVote);
 }
 
 function displayResults(poll) {
-
+  // Create 
   let resultsDiv = document.getElementById('results');
 
-  let maxVotes = 0;
-  let totalVotes = 0;
-  for (let count of poll.votes) {
-    maxVotes = max(count, maxVotes);
-    totalVotes += count;
-  }
+  // Get the number of votes that the most voted option has.
+  let maxVotes = poll.votes.reduce((a, b) => (a > b ? a : b));
 
-  for(const [index, option] of Object.entries(poll.options)) {
+  // Get the total number of votes.
+  let totalVotes = poll.votes.reduce((a, b) => (a + b));
+
+  for (const [index, option] of Object.entries(poll.options)) {
     let count = poll.votes[index];
     let width = map(count, 0, maxVotes, 0, 100);
-    if (firstRender) {
-      //Container div per option
-      let optionDiv = document.createElement('div');
-      optionDiv.classList.add('option');
-      // Add option text on a separate line, so it can get any length
-      let optionText = document.createElement('span');
-      optionText.innerHTML = option;
-      optionDiv.append(optionText);
-      resultsDiv.append(optionDiv);
-
-      //Add option progress bar
-      let progressBar = document.createElement('div');
-      progressBar.id = 'progressBar_' + index;
-      progressBar.classList.add('progressBar');
-      progressBar.style.setProperty('width', width + '%');
-      progressBar.innerHTML = Math.round(count/totalVotes*100) + '%';
-      resultsDiv.append(progressBar);
-    } else {
-      let progressBar = document.getElementById("progressBar_" + index);
-      progressBar.style.setProperty('width', width + '%');
-      progressBar.innerHTML = Math.round(count/totalVotes*100) + '%';
-    }
-  }
 
+    if (firstRender)
+      doTheFirstRender(index, option, resultsDiv);
+
+    // Get the progress bar element
+    let progressBar = document.getElementById("progressBar_" + index);
+    // Set the width
+    progressBar.style.setProperty('width', width + '%');
+    // Set the text
+    progressBar.innerHTML = Math.round(count / totalVotes * 100) + '%';
+  }
```

Comment:
How would you feel about refactoring these to use p5.js functions: `select()`, `style()`, `html()`. I don't feel strongly but as of now other parts of the application are using p5 dom functions so might be good to be consistent?